test/provider.js

```diff
@@ -65,14 +65,20 @@ describe('Provider', function () {
               headers: notification.headers(),
               body: notification.compile(),
             };
+            const device = 'abcd1234';
             expect(fakes.client.write).to.be.calledOnce;
-            expect(fakes.client.write).to.be.calledWith(builtNotification, 'abcd1234');
+            expect(fakes.client.write).to.be.calledWith(
+              builtNotification,
+              device,
+              'device',
+              'post'
+            );
           });
         });
 
         it('does not pass the array index to writer', function () {
           return provider.send(notificationDouble(), 'abcd1234').then(function () {
-            expect(fakes.client.write.firstCall.args[2]).to.be.undefined;
+            expect(fakes.client.write.firstCall.args[4]).to.be.undefined;
```

Comment:
This is because I changed the number of arguments, but still left `count` as the last arg